librz/analysis/esil/esil_trace.c

```diff
@@ -244,10 +244,10 @@ static int trace_hook_mem_write(RzAnalysisEsil *esil, ut64 addr, const ut8 *buf,
  * \return RzILTraceInstruction *, instruction trace at index
  */
 RZ_API RzILTraceInstruction *rz_analysis_esil_get_instruction_trace(RzAnalysisEsilTrace *etrace, int idx) {
```

Comment:
Please add `rz_return_val_if_fail(etrace, NULL)` at the beginning. Also `RZ_NONNULL` to `etrace`. Also `RZ_BORROW` to the return.